scheduler/scheduler.go

```diff
@@ -30,7 +30,7 @@ import (
 	"path"
 )
 
-type leaderFunc func(<-chan interface{}, coordclient.Connection, dao.ControlPlane, string, int)
+type leaderFunc func(<-chan interface{}, coordclient.Connection, dao.ControlPlane, *facade.Facade, string, int)
```

Comment:
why do we need connections to the dao and facade?  Is it possible to pick one here?